test/unit/core/metafunction/transformation_trait_or_test.cpp

```diff
@@ -0,0 +1,67 @@
+// ============================================================================
+//                 SeqAn - The Library for Sequence Analysis
+// ============================================================================
+//
+// Copyright (c) 2006-2018, Knut Reinert & Freie Universitaet Berlin
+// Copyright (c) 2016-2018, Knut Reinert & MPI Molekulare Genetik
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+#include <gtest/gtest.h>
+
+#include <seqan3/core/metafunction/transformation_trait_or.hpp>
+
+using namespace seqan3;
+
+struct A
+{
+    using type = int;
+};
+
+struct B;
+
+struct C
+{};
+
+struct D
+{
+    static constexpr int type = 6;
+};
+
+TEST(transformation_trait_or, transformation_trait_or)
+{
+    using a_type = detail::transformation_trait_or_t<A, void>;
+    using b_transformation_trait_or = detail::transformation_trait_or_t<B, void>;
+    using c_transformation_trait_or = detail::transformation_trait_or_t<C, double>;
+    using d_transformation_trait_or = detail::transformation_trait_or<D, B>::type;
+
+    EXPECT_TRUE((std::is_same_v<a_type, int>));
+    EXPECT_TRUE((std::is_same_v<b_transformation_trait_or, void>));
+    EXPECT_TRUE((std::is_same_v<c_transformation_trait_or, double>));
+    EXPECT_TRUE((std::is_same_v<d_transformation_trait_or, B>));
+}
```

Comment:
Maybe use this directly as the snippet? :thinking:

Reply:
I don't like `EXPECT_TRUE` in snippets.